bokehjs/src/coffee/models/tools/actions/save_tool.ts

```diff
@@ -0,0 +1,19 @@
+import {ActionTool, ActionToolView} from "./action_tool"
+
+export class SaveToolView extends ActionToolView {
+
+  model: SaveTool
+
+  doit(): void {
+    this.plot_view.save("bokeh_plot")
+  }
+}
+
+export class SaveTool extends ActionTool {
+  tool_name = "Save"
+  icon = "bk-tool-icon-save"
+}
+
+SaveTool.prototype.type = "SaveTool"
+
+SaveTool.prototype.default_view = SaveToolView
```

Comment:
All of these actions are very similar I wonder if they could be created generatively like markers

Reply:
It can be shortened, but without macros there isn't much that can be done here. Though I would consider getting rid of `default_view` (at least by default) and deriving it in `base`.